tests/pds/dataservice_test.go

```diff
@@ -136,128 +133,40 @@ var _ = BeforeSuite(func() {
-						Expect(err).NotTo(HaveOccurred())
-						logrus.Infof("filesystem used %v ", config.Spec.StorageOptions.Filesystem)
-						logrus.Infof("storage replicas used %v ", config.Spec.StorageOptions.Replicas)
-						logrus.Infof("cpu requests used %v ", config.Spec.Resources.Requests.CPU)
-						logrus.Infof("memory requests used %v ", config.Spec.Resources.Requests.Memory)
-						logrus.Infof("storage requests used %v ", config.Spec.Resources.Requests.Storage)
-						logrus.Infof("No of nodes requested %v ", config.Spec.Nodes)
-						logrus.Infof("volume group %v ", storageOp.VolumeGroup)
-
-						Expect(resourceTemp.Resources.Requests.CPU).Should(Equal(config.Spec.Resources.Requests.CPU))
-						Expect(resourceTemp.Resources.Requests.Memory).Should(Equal(config.Spec.Resources.Requests.Memory))
-						Expect(resourceTemp.Resources.Requests.Storage).Should(Equal(config.Spec.Resources.Requests.Storage))
-						Expect(resourceTemp.Resources.Limits.CPU).Should(Equal(config.Spec.Resources.Limits.CPU))
-						Expect(resourceTemp.Resources.Limits.Memory).Should(Equal(config.Spec.Resources.Limits.Memory))
-						repl, err := strconv.Atoi(config.Spec.StorageOptions.Replicas)
-						Expect(err).NotTo(HaveOccurred())
-						Expect(storageOp.Replicas).Should(Equal(int32(repl)))
-						Expect(storageOp.Filesystem).Should(Equal(config.Spec.StorageOptions.Filesystem))
-						Expect(config.Spec.Nodes).Should(Equal(replicas))
-						for version, build := range dataServiceVersionBuildMap {
-							Expect(config.Spec.Version).Should(Equal(version + "-" + build[index]))
-						}
-
-					}
-				}
-			})
-		})
-
-		Step("Running Workloads before scaling up of dataservices ", func() {
-			for ds, deployment := range deployments {
-				for index := range deployment {
-					if ds == postgresql {
-						deploymentName := "pgload"
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds, deployment[index].GetId(), "100", "1", deploymentName, namespace)
-						Expect(err).NotTo(HaveOccurred())
-					}
-					if ds == rabbitmq {
-						deploymentName := "rmq"
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds, deployment[index].GetId(), "", "", deploymentName, namespace)
-						Expect(err).NotTo(HaveOccurred())
-					}
-					if ds == redis {
-						deploymentName := "redisbench"
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds, deployment[index].GetId(), "", "", deploymentName, namespace)
-						Expect(err).NotTo(HaveOccurred())
-					}
-					if ds == cassandra {
-						deploymentName := "cassandra-stress"
-						pod, dep, err = pdslib.CreateDataServiceWorkloads(ds, deployment[index].GetId(), "", "", deploymentName, namespace)
-						Expect(err).NotTo(HaveOccurred())
-					}
-				}
-			}
-		})
-
-		Step("Update the data service patch versions", func() {
+func UpgradeDataService(oldVersion string, oldImage string, dsVersion string, dsBuild string) {
+
+	Step("Deploy Data Services", func() {
+		logrus.Infof("Version/Build: %v %v", oldVersion, oldImage)
```

Comment:
can we move the deployment and deletion of services to justBeforeEach and JustAfterEach

Reply:
there are other specs which use the same function with different variables, will check with Madan if it can be done

Reply:
As mentioned, the deployment ids, names etc will not be accessible in JustAfterEach for cleanup, and there will be a bit of code repetition if we move the deploy and teardown steps to JustBeforeEach, JustAfterEach